recipes/data-prep-kit/end-2-end-code-data-prep-finetuning.ipynb

```diff
@@ -0,0 +1,869 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "NbF_Zw3KBazf"
+   },
+   "source": [
+    "# **Demo on building data prep pipeline for model fine tuning** "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"https://colab.research.google.com/github/IBM/data-prep-kit/blob/dev/examples/notebooks/code/sample-notebook.ipynb\">\n",
+    "  <img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/>\n",
+    "</a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "_-NOkuTxiP7r",
+    "outputId": "043f32fc-c476-433e-86b6-d7e9abd4d285"
+   },
+   "source": [
+    "This demo notebook shows how to use [data-prep-kit](https://github.com/IBM/data-prep-kit) to build a data preparation pipeline that can be used for fine tuning or extended pre-training. We will discuss the various data preparation steps to process raw data (code repositories), tokenise it that can then be fine tuned using any popular code models. We will also discuss a novel recipe for semantic ordering of files in a repository which has shown to enhance model training. Please see our [paper](https://arxiv.org/abs/2407.13739) here for more details. For this demo, we will use the [codeparrot/github-code](https://huggingface.co/datasets/codeparrot/github-code) dataset hosted on Hugging Face datasets. \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "Install data-prep-toolkit and datasets library. This notebook requires atleast 8 cpus. \n",
+    "To run on google colab, it is recommended to change the runtime to TPUs to get the required number of cpus.\n"
```

Comment:
Can this notebook be reworked to drop this requirement. At best, you won't get 8cpus for the testing (at least w/o some additional config, which I don't know how to do off the top of my head).

Reply:
We have updated to 4. There is guidance in the notebook to select TPUs and these nodes can be then made available via Colab.

Reply:
Not in testing though. You can only get 2 cpus for the test suite